iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/symbols/TupleBlockSymbol.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.kubernetes.symbols;
+
+import javax.annotation.Nullable;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.common.yaml.tree.MappingTree;
+import org.sonar.iac.common.yaml.tree.TupleTree;
+import org.sonar.iac.common.yaml.tree.YamlTree;
+
+public class TupleBlockSymbol extends BlockSymbol<TupleBlockSymbol, MappingTree> {
+
+
+  private final HasTextRange toHighlight;
+
+  protected TupleBlockSymbol(CheckContext ctx, @Nullable HasTextRange toHighlight, MappingTree block, String key, BlockSymbol<?,?> parent) {
+    super(ctx, block, key, parent);
+    this.toHighlight = toHighlight;
+  }
+
+  public static TupleBlockSymbol fromPresent(CheckContext ctx, YamlTree tree, String key, BlockSymbol<?,?> parent) {
+    if (tree instanceof TupleTree && ((TupleTree) tree).value() instanceof MappingTree) {
```

Comment:
We have this pattern in multiple places: either the tree is what we expect it to be or we consider it to be absent.
I wonder if we miss a distinction between:
* it's really absent
* it's there, but not as we expect (not the type we expect)

I wonder if that could lead to FPs in the end.